base/abstractarray.jl

```diff
@@ -343,6 +343,38 @@ function first(itr)
     x[1]
 end
 
+"""
+    first(predicate, coll)
+
+Get the first element of `coll` satisfying `predicate` wrapped in [`Some`](@ref).
+
+If no element of `coll` satisfies `predicate`, return `nothing`.
```

Comment:
This is interesting — and deviates from the existing `first` which would throw an error.  This gives me a bit of pause, but at a minimum it means that we cannot combine these docstrings:

```Julia
julia> first(_->true, [])

julia> first([])
```

Reply:
Good point on the docstrings.